corehq/apps/accounting/utils.py

```diff
@@ -354,3 +355,20 @@ def clear_plan_version_cache():
     from corehq.apps.accounting.models import SoftwarePlan
     for software_plan in SoftwarePlan.objects.all():
         SoftwarePlan.get_version.clear(software_plan)
+
+
+def get_role_edition(role_slug):
+    from corehq.apps.accounting.models import SoftwarePlanVersion
+    all_editions = SoftwarePlanVersion.objects.filter(
+        role__slug=role_slug).distinct(
+        'plan__edition').values_list('plan__edition', flat=True)
+
+    if len(all_editions) == 1:
+        return all_editions[0]
+
+    def _count_edition(edition):
+        return SoftwarePlanVersion.objects.filter(
+            role__slug=role_slug, plan__edition=edition).count()
+
+    edition_counts = {e: _count_edition(e) for e in all_editions}
+    return max(edition_counts.items(), key=operator.itemgetter(1))[0]
```

Comment:
I think these two lines can be simplified to
```py
return max(all_editions, key=_count_edition)
```
Also, this executes a separate database query for every item in `all_editions`, which will not perform well if there are many editions.

Reply:
thanks. yes, I know that's the case re: separate db query. I tried finding a workaround by adding `.annotate(Count('plan__edition'))` to the original query but couldn't get that to work. not sure if it's the django version we're on... in any case there shouldn't be too many editions anyway.

Reply:
I had been thinking this function (`get_role_edition`) was called as part of user auth, which was concerning. I see now that it's only used by a management command. Would it make sense to move it into the management command module so no one will be tempted to use it and it will get deleted with that module (if that ever happens)?

Reply:
done :)